File: backend/models/exchange_rate.py
from datetime import datetime

from pydantic import BaseModel, Field


class ExchangeRateRecord(BaseModel):
    """汇率历史记录"""

    id: int
    rate: float
    timestamp: datetime


class ExchangeRateStatistics(BaseModel):
    """汇率统计"""

    min: float = Field(..., description="最低汇率")
    max: float = Field(..., description="最高汇率")
    avg: float = Field(..., description="均价")
    count: int = Field(..., description="数据量")
    std: float = Field(..., description="标准差")


class ExchangeRateChartPoint(BaseModel):
    """走势图数据点"""

    timestamp: datetime = Field(..., description="记录时间")
    rate: float = Field(..., description="汇率")


class ExchangeRateTrend(BaseModel):
    """汇率趋势"""

    slope: float = Field(..., description="斜率")
    direction: str = Field(..., description="趋势方向：up/down/stable")


class ExchangeRateDashboardItem(BaseModel):
    """汇率仪表盘项"""

    record_count: int = Field(default=0, description="总记录数")
    latest_rate: float | None = Field(default=None, description="最新汇率")
    latest_time: datetime | None = Field(default=None, description="最新记录时间")
    today_high: float | None = Field(default=None, description="今日最高")
    today_low: float | None = Field(default=None, description="今日最低")
    data_freshness_seconds: int | None = Field(
        default=None, description="距最新数据秒数"
    )

File: frontend/src/api/modules/exchange-rate.ts
import request from '@/api/request'

export interface ExchangeRateStatistics {
  min: number
  max: number
  avg: number
  count: number
  std: number
}

export interface ExchangeRateTrend {
  slope: number
  direction: 'up' | 'down' | 'stable'
}

export interface ExchangeRateChartPoint {
  timestamp: string
  rate: number
}

export interface ExchangeRateDashboard {
  record_count: number
  latest_rate: number
  latest_time: string | null
  today_high: number | null
  today_low: number | null
  data_freshness_seconds: number | null
}

export interface ExchangeRateRecord {
  timestamp: string
  rate: number
}

export const exchangeRateApi = {
  getCount() {
    return request.get<number>('/exchange-rate/count')
  },

  getStatistics(params?: { hours?: number; start_date?: string; end_date?: string }) {
    return request.get<ExchangeRateStatistics>('/exchange-rate/statistics', { params })
  },

  getTrend(params?: { hours?: number; start_date?: string; end_date?: string }) {
    return request.get<ExchangeRateTrend>('/exchange-rate/trend', { params })
  },

  getDashboard(params?: { hours?: number; start_date?: string; end_date?: string }) {
    return request.get<ExchangeRateDashboard>('/exchange-rate/dashboard', { params })
  },

  getChart(params?: { hours?: number; start_date?: string; end_date?: string }) {
    return request.get<ExchangeRateChartPoint[]>('/exchange-rate/chart', { params })
  },

  getRecent(params?: { hours?: number; limit?: number }) {
    return request.get<ExchangeRateRecord[]>('/exchange-rate/recent', { params })
  },
}
